### PR TITLE
Refactors join community

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9345

## Description of Changes
- Refactors old endpoint
- Refactors model utils to avoid passing DB models and resolving cycles 

## Test Plan
- There are changes affecting how we create a community (@mzparacha to review UI changes).. but we should be able to create one
- Make sure you can join a community, testing all base options
- Unit test needs review to validate all options

## Deployment Plan

1. 

## Other Considerations

- Discuss TODOs with team

### CODE REVIEW

**Comment by Rotorsoft** [comment]
@timolegros this is only failing the join community test we discussed... let's figure out how to configure the community to pass this

**Review by timolegros — DISMISSED** [review]
Seems to mostly maintain existing functionality though I encountered this bug ~~which seems to exist on master well~~: https://github.com/hicommonwealth/commonwealth/issues/9362. ~~Approving pre-emptively just need to modify the tests per~~ ~~my Slack DM.~~

File: libs/schemas/src/commands/community.schemas.ts [hunk 1]
```diff
@@ -275,3 +275,18 @@ export const RefreshCommunityMemberships = {
     updated: z.number(),
   }),
 };
+
+export const JoinCommunity = {
+  input: z.object({
+    community_id: z.string(),
+  }),
+  output: z.object({
+    community_id: z.string(),
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/schemas/src/commands/community.schemas.ts:284]: We should probably create a ticket with all API schemas we should consider for this type of refactoring. Extra response details won't hurt unless there is sensitive data included, but we also need to consider what our client is using right now from the responses.

File: packages/commonwealth/client/scripts/state/api/communities/createCommunity.ts [hunk 2]
```diff
@@ -45,14 +40,7 @@ export const buildCreateCommunityInput = ({
 
 const useCreateCommunityMutation = () => {
   return trpc.community.createCommunity.useMutation({
-    onSuccess: async (output) => {
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/state/api/communities/createCommunity.ts:48]: This seems redundant in the client... we are already linking the admin address to the community inside CreateCommunity.command

File: libs/schemas/src/commands/community.schemas.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (timolegros) [reply libs/schemas/src/commands/community.schemas.ts:284]: https://github.com/hicommonwealth/commonwealth/issues/9378

**Review by timolegros — APPROVED** [review]

File: libs/model/src/middleware/authorization.ts [hunk 3]
```diff
@@ -149,6 +149,8 @@ async function buildAuth(
       address: actor.address,
       community_id: auth.community_id,
       role: { [Op.in]: roles },
+      verified: { [Op.ne]: null },
+      // TODO: check verification token expiration
```
Comment (mzparacha) [inline libs/model/src/middleware/authorization.ts:153]: still todo?

Comment (Rotorsoft) [reply libs/model/src/middleware/authorization.ts:153]: yes, this is linked to the API auth work @timolegros ?

**Review by mzparacha — APPROVED** [review]
Tried creating solana/ethereum/cosmos communities and joining solana/ethereum/cosmos/substrate/evmos/injective communities with multiple addresses. All worked well. Thanks for all the changes 🎉

**Pull request merged**